services/reddit/user-karma.tester.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const { isMetric } = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('user-karma (valid - link)')
+  .get('/link/user_simulator.json')
+  .expectBadge({
+    label: "user_simulator's karma (link)",
+    message: isMetric,
+  })
+
+t.create('user-karma (valid - comment')
+  .get('/comment/user_simulator.json')
+  .expectBadge({
+    label: "user_simulator's karma (comment)",
+    message: isMetric,
+  })
+
+t.create('user-karma (valid - combined)')
+  .get('/combined/user_simulator.json')
+  .expectBadge({
+    label: "user_simulator's karma",
+    message: isMetric,
+  })
+
+t.create('user-karma (non-existing user)')
+  .get('/combined/thisuserdoesnotexistandhopefullyneverwill.json')
+  .expectBadge({
+    label: 'reddit karma',
+    message: 'user not found',
+  })
+
+t.create('user-karma (link - math check)')
+  .get('/link/user_simulator.json')
+  .intercept(nock =>
+    nock('https://www.reddit.com/u')
+      .get('/user_simulator/about.json')
+      .reply(200, { kind: 't2', data: { link_karma: 20, comment_karma: 80 } })
+  )
+  .expectBadge({
+    label: "user_simulator's karma (link)",
+    message: '20',
+  })
```

Comment:
a common pattern many of a service classes use is to leverage a `transform` function that accepts the raw json response data and returns the corresponding value. one of the benefits of that approach is that it makes it simpler to test logic that's extracting/transforming with a simple unit test instead of having to use a full service test (which are primarily used as integration tests with the myriad upstream platforms/providers we integrate with).

don't feel like you have to change that here, but sharing for awareness in case you want to/for any potential future use cases

Reply:
Noted, I will definitely look at doing that in the future.